src/qibo/models/encodings.py

```diff
@@ -528,28 +528,24 @@ def entangling_layer(
         circuit = Circuit(nqubits, **kwargs)
 
         if architecture == "diagonal":
-            circuit.add(
-                _parametrized_two_qubit_gate(gate, qubit, qubit + 1, parameters)
-                for qubit in range(nqubits - 1)
-            )
+            qubits = range(nqubits - 1)
         elif architecture == "even_layer":
-            circuit.add(
-                _parametrized_two_qubit_gate(gate, qubit, qubit + 1, parameters)
-                for qubit in range(0, nqubits - 1, 2)
-            )
+            qubits = range(0, nqubits - 1, 2)
         elif architecture == "odd_layer":
-            circuit.add(
-                _parametrized_two_qubit_gate(gate, qubit, qubit + 1, parameters)
-                for qubit in range(1, nqubits - 1, 2)
-            )
+            qubits = range(1, nqubits - 1, 2)
         else:
+            qubits = range(0, nqubits - 1, 2)
+            qubits_2 = range(1, nqubits - 1, 2)
```

Comment:
```suggestion
            qubits = tuple(range(0, nqubits - 1, 2)) + tuple(range(1, nqubits - 1, 2))
```
this should lift the need of the following ``architecture == "shifted"`` check I believe